### ISSUE TITLE
Add support for source maps?

### ISSUE THREAD

**Issue by Aeon** [csnover/TraceKit#31]
Cloning over from https://github.com/occ/TraceKit/issues/61:

-----
Adding support for getting cleaner stack traces from minified JS files via source maps would be pretty amazing.

Can leverage mozilla/source-map project for [decomposing sources in ES5-compliant browsers](https://github.com/mozilla/source-map#consuming-a-source-map)

**Comment by niemyjski** [comment]
Yes, I would like this to be a plugin so those who need it can buy into it. This is something we will be working on in the near future. I'd also like source maps. The last time I looked into it, they only supported amd, is this still the case?

**Comment by auchenberg** [comment]
Inspiration https://github.com/stacktracejs/stacktrace-gps

**Comment by niemyjski** [comment]
Yes, I really really want this and could use this sooner than later. I need to start working on the plugin system. I want to have source maps be an optional plugins because mozilla source maps is huge (30KB).

**Comment by mightyiam** [comment]
:+1: from https://github.com/jquery/qunit/issues/590

**Comment by davidomid** [comment]
Was there any further progress with this? Source maps would be incredibly useful. I'm happy to take a stab at a possible implementation if nobody else has a viable one yet?

**Comment by niemyjski** [comment]
We were trying to change up tracekit to be more plugable by that effort stalled a bit. If you want to take a stab at this I'd be more than happy to work with you. It just needs to be optional as the source map lib is an extra 30KB

**Comment by vivekmarakana** [comment]
Can I take this up? I've already created a plugin for the same.

**Comment by niemyjski** [comment]
yeah, please do! It would be greatly appreciated

**Comment by vivekmarakana** [comment]
Great. will open a pull request soon.

**Comment by ncoden** [comment]
Hey @vivekmarakana. How is it going ?

**Comment by adrienbu** [comment]
Any news on this ?

**Comment by niemyjski** [comment]
I haven't heard anything on this. Would you mind jumping in and submit a pr
for this.

>

**Comment by adrienbu** [comment]
I'll have a try but I cannot promise anything !
